presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/csv/CsvRowEncoder.java

```diff
@@ -0,0 +1,151 @@
+
+public class CsvRowEncoder
+        extends AbstractRowEncoder
+{
+    private static final Set<Type> SUPPORTED_PRIMITIVE_TYPES = ImmutableSet.of(
+            BOOLEAN, TINYINT, SMALLINT, INTEGER, BIGINT, DOUBLE, REAL);
+
+    public static final String NAME = "csv";
+
+    private final int size;
+    private String[] row;
+
+    public CsvRowEncoder(ConnectorSession session, List<EncoderColumnHandle> columnHandles)
+    {
+        super(session, columnHandles);
+        this.size = columnHandles.size();
+        this.row = new String[this.size];
+    }
+
+    @Override
+    protected List<EncoderColumnHandle> validateColumns(List<EncoderColumnHandle> columnHandles)
+    {
+        for (EncoderColumnHandle columnHandle : columnHandles) {
+            checkArgument(columnHandle.getFormatHint() == null, "unexpected format hint '%s' defined for column '%s'", columnHandle.getFormatHint(), columnHandle.getName());
+            checkArgument(columnHandle.getDataFormat() == null, "unexpected data format '%s' defined for column '%s'", columnHandle.getDataFormat(), columnHandle.getName());
+
+            checkArgument(isSupportedType(columnHandle.getType()), "unsupported column type '%s' for column '%s'", columnHandle.getType(), columnHandle.getName());
+        }
+        return ImmutableList.copyOf(columnHandles);
+    }
+
+    private boolean isSupportedType(Type type)
+    {
+        return isVarcharType(type) || SUPPORTED_PRIMITIVE_TYPES.contains(type);
+    }
+
+    @Override
+    protected void appendNullValue()
+    {
+        row[currentColumnIndex] = null;
+    }
+
+    @Override
+    protected void appendLong(long value)
+    {
+        row[currentColumnIndex] = Long.toString(value);
+    }
+
+    @Override
+    protected void appendInt(int value)
+    {
+        row[currentColumnIndex] = Integer.toString(value);
+    }
+
+    @Override
+    protected void appendShort(short value)
+    {
+        row[currentColumnIndex] = Short.toString(value);
+    }
+
+    @Override
+    protected void appendByte(byte value)
+    {
+        row[currentColumnIndex] = Byte.toString(value);
+    }
+
+    @Override
+    protected void appendDouble(double value)
+    {
+        row[currentColumnIndex] = Double.toString(value);
+    }
+
+    @Override
+    protected void appendFloat(float value)
+    {
+        row[currentColumnIndex] = Float.toString(value);
+    }
+
+    @Override
+    protected void appendBoolean(boolean value)
+    {
+        row[currentColumnIndex] = Boolean.toString(value);
+    }
+
+    @Override
+    protected void appendString(String value)
+    {
+        row[currentColumnIndex] = value;
+    }
+
+    @Override
+    public byte[] toByteArray()
+    {
+        try (ByteArrayOutputStream byteArrayOuts = new ByteArrayOutputStream();
+                OutputStreamWriter outsWriter = new OutputStreamWriter(byteArrayOuts, StandardCharsets.UTF_8);
+                CSVWriter writer = new CSVWriter(outsWriter, ',', '"', "")) {
```

Comment:
do we support overriding special characters on read? If so we should also support them on write.

Reply:
You mean like using something other than a comma to separate the values?

Reply:
Yep. Or different quote char.

Reply:
I don't see anything about that in the docs for read so I'm gonna say no it doesn't. Would be a nice feature tho